mewarpx/tests/test_emission.py

```diff
@@ -749,7 +750,91 @@ def test_arbitrary_distribution_emitter():
-    assert np.allclose(ion_xavg[10:502], sin_xavg[10:502], rtol=0.06)
-    assert np.allclose(ion_zavg[10:502], sin_zavg[10:502], rtol=0.06)
+    slices = np.s_[10:-10]
+
+    assert np.allclose(electron_xavg[slices], sin_xavg[slices], rtol=0.06)
+    assert np.allclose(electron_zavg[slices], sin_zavg[slices], rtol=0.06)
+    assert np.allclose(ion_xavg[slices], sin_xavg[slices], rtol=0.06)
+    assert np.allclose(ion_zavg[slices], sin_zavg[slices], rtol=0.06)
+
+
+def test_arbitrary_distribution_emitter_paraboloid():
+    """Seeds the simulation with a parabolic cylinder distribution"""
+    name = "arbitrary_distribution_emitter_paraboloid"
+    # Include a random run number to allow parallel runs to not collide.  Using
+    # python randint prevents collisions due to numpy rseed below
+    testing_util.initialize_testingdir(name)
+
+    # Initialize each run with consistent, randomly-chosen, rseed. Use a random
+    # seed instead for initial dataframe generation.
+    np.random.seed(11874889)
+
+    NX = NZ = 512
+    NPPC = 10
+
+    run = diode_setup.DiodeRun_V1(
+        GEOM_STR='XZ', PERIOD=0.8e-06 * NX, D_CA=0.8e-06 * NZ, DT=1e-12,
+        TOTAL_TIMESTEPS=1, DIAG_STEPS=1
+    )
+
+    run.setup_run(init_inert_gas=True, init_scraper=False, init_injectors=False)
+
+    # generate scaled down plasma density used for seeding
+    z = np.linspace(0.5, NZ - 0.5, 256)
+    x = np.linspace(0.5, NX - 0.5, 256)
+
+    zz, xx = np.meshgrid(z, x)
+
+    plasma_density = 0.5e13 * (-1 / (256**2) * (128 * zz  + (xx - 256)**2) + 2)
+
+    # generate full resolution plasma density used as test's reference
+    z = np.linspace(0, NZ, NZ + 1)
+    x = np.linspace(0, NX, NX + 1)
+    zz, xx = np.meshgrid(z, x)
+    ref_plasma_density = 0.5e13 * (-1 / (256**2) * (128 * zz  + (xx - 256)**2) + 2)
+
+    emitter = emission.ArbitraryDistributionVolumeEmitter(
+        d_grid=plasma_density,
+        T=1550, zmin=0, zmax=run.D_CA,
+    )
+
+    emission.PlasmaInjector(
+        emitter=emitter, species1=run.electrons, species2=run.ions,
+        npart=NX*NZ*NPPC
+    )
+
+    diags.FieldDiagnostic(
+        diag_steps=1, style='roelof', save_pdf=False,
+        plot=True,
+    )
+
+    diags.TextDiag(
+        diag_steps=1, preset_string='perfdebug'
+    )
+
+    run.init_warpx()
+    mwxrun.simulation.step(2)
+
+    electron_density = np.load(os.path.join(
+        os.curdir, "diags", "fields",
+        "electrons_particle_density_0000000002.npy"
+    ))
+    electron_xavg = np.mean(electron_density, axis=0)
+    electron_zavg = np.mean(electron_density, axis=1)
+
+    ion_density = np.load(os.path.join(
+        os.curdir, "diags", "fields",
+        "ar_ions_particle_density_0000000002.npy"
+    ))
+    ion_xavg = np.mean(ion_density, axis=0)
+    ion_zavg = np.mean(ion_density, axis=1)
+
+    sin_xavg = np.mean(ref_plasma_density, axis=0)
+    sin_zavg = np.mean(ref_plasma_density, axis=1)
```

Comment:
rename these vars

Reply:
Should I include the numpy files for the sin density distribution as well, or just the parabolic cylinder?